File: backend/orchestrator-prompt/system/product_promotion_strategy.py
# 製品の販促戦略作成
def get_product_promotion_strategy_system_message(
    product_description: str,
    target_market: str,
    differentiation_point: str,
    promotion_tools: str,
    sales_channels: str,
) -> str:
    PROMOTION_STRATEGY_SYSTEM_MESSAGE = f"""\
    ### 役割
    あなたは優秀なコンサルタントです。
    新製品の販売促進のための戦略を一緒に考えてほしいです。

    ### 指示
    以下#入力欄の情報をもとに、販売促進戦略を出力してください。

    ### 入力欄
    製品説明（製品の特長などを記入）：{product_description}
    主なターゲット市場（顧客層）：{target_market}
    競合との差別化ポイント：{differentiation_point}
    考えている販促ツール：{promotion_tools}
    主な販売チャネル：{sales_channels}

    ### 出力形式
    以下の構成で販売促進戦略を作成してください：
    1. **戦略概要**
    2. **ターゲット分析**
    3. **差別化戦略**
    4. **販促施策の詳細**
    5. **チャネル戦略**
    6. **実行計画とタイムライン**
    7. **成功指標（KPI）**
    8. **予算配分の推奨**

    ### 制約条件
    - 実践的で具体的な提案を行う
    - ROIを意識した効果的な施策を提案
    - 段階的な実施計画を含める
    - 測定可能な成果指標を設定
    """
    return PROMOTION_STRATEGY_SYSTEM_MESSAGE


def get_product_promotion_strategy_message(
    product_description: str,
    target_market: str,
    differentiation_point: str,
    promotion_tools: str,
    sales_channels: str,
):
    messages = [
        {
            "role": "user",
            "content": get_product_promotion_strategy_system_message(
                product_description,
                target_market,
                differentiation_point,
                promotion_tools,
                sales_channels,
            ),
        },
    ]
    return messages

File: frontend/src/app/(dashboard)/product-promotion-strategy/_actions/productPromotionStrategy.ts
'use server';

import { productPromotionStrategyDB } from '@/app/_db/product-promotion-strategy';
import { LLMserviceBackEndLog } from '@/app/_types/logs/llm-services';
import { getCurrentUser } from '@/app/_utils/auth';
import { getMessage } from '@/app/_utils/message';
import { useCaseAzureFunctions } from '../../../../../azure-functions';
import { productPromotionStrategyContainer } from '../../../../../cosmos';

type ProductPromotionStrategyResponse = {
  result: string;
  success: boolean;
  log: LLMserviceBackEndLog<'productPromotionStrategy'>;
};

type ProductPromotionStrategyErrorResponse = {
  error: string;
};

export async function createProductPromotionStrategy(
  id: string,
  productDescription: string,
  targetMarket: string,
  differentiationPoint: string,
  promotionTools: string,
  salesChannels: string
): Promise<ProductPromotionStrategyResponse | ProductPromotionStrategyErrorResponse> {
  const user = await getCurrentUser();
  try {
    const response = await useCaseAzureFunctions.sendJson<
      {
        productDescription: string;
        targetMarket: string;
        differentiationPoint: string;
        promotionTools: string;
        salesChannels: string;
      },
      ProductPromotionStrategyResponse
    >('product-promotion-strategy', 'POST', {
      productDescription,
      targetMarket,
      differentiationPoint,
      promotionTools,
      salesChannels,
    });

    // CosmosDBにログを保存
    await productPromotionStrategyDB(productPromotionStrategyContainer, {
      id,
      userId: user.id,
      userName: user.name,
      userEmail: user.email,
      userDepartmentName: user.departmentName,
      productDescription,
      targetMarket,
      differentiationPoint,
      promotionTools,
      salesChannels,
      createdAt: new Date(),
      outputForm: response.result,
      log: response.log,
    });

    return response;
  } catch (error) {
    console.error('Error creating product promotion strategy:', error);
    return {
      error: error instanceof Error ? error.message : getMessage('E_F_00110', '製品の拡販戦略'),
    };
  }
}
